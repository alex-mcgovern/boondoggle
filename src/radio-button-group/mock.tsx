import type * as React from "react";

import type { RadioButton } from "../radio-button";

import { LOREM } from "../../mocks/LOREM.mock";

export const RADIO_BUTTONS_MOCK: Array<
	React.ComponentProps<typeof RadioButton>
> = [
	{
		description: LOREM.text_xxs,
		name: "radio_item_1",
		title: "Radio item 1",
		value: "radio_item_1",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_2",
		title: "Radio item 2",
		value: "radio_item_2",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_3",
		title: "Radio item 3",
		value: "radio_item_3",
	},
];
