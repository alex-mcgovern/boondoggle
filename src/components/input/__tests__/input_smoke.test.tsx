/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";

import { Input } from "..";
import { LOREM } from "../../../../mocks/LOREM.mock";

import type { InputProps } from "..";

const PROPS: InputProps = {
	name: LOREM.name(),
	placeholder: LOREM.placeholder(),
};

const renderComponent = ({ ...props }: InputProps) => {
	return render(<Input {...props} />);
};

describe("<Input />", () => {
	describe("Basic smoke tests", () => {
		test("should render without throwing", () => {
			const { getByRole } = renderComponent(PROPS);

			expect(getByRole("textbox")).not.toBeNull();
		});
	});
});
