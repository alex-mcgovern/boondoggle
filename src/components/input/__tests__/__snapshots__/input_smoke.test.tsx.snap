// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Basic smoke tests should match snapshot 1`] = `
<input
  class="styles_getInputStyles__1i5hpgs6 styles__1i5hpgs5 get_sprinkles_width_100%_mobile__eiij7bn1 a11y_a11yDisabled__r8he5t1 styles_getInputStyles_hasBorder_true__1i5hpgs8 styles_hasBorder_true__1i5hpgs4 styles_getInputStyles_size_md__1i5hpgsa styles_size_md__1i5hpgs1 element_size_elementFontSize_md__1f6zrv4 element_size__1f6zrv1 get_sprinkles_fontSize_body_md__eiij7b21 get_sprinkles_lineHeight_body_md__eiij7b3n element_size_elementPadding_md__1f6zrva element_size__1f6zrv7 get_sprinkles_paddingLeft_space_3_mobile__eiij7bid get_sprinkles_paddingRight_space_3_mobile__eiij7bjy get_sprinkles_paddingTop_space_2_mobile__eiij7blg get_sprinkles_paddingBottom_space_2_mobile__eiij7bgp"
  name="Name ipsum"
  placeholder="Placeholder ipsum"
  value=""
/>
`;
